src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs

```diff
@@ -142,94 +163,98 @@ public void ConstructorAndFinalizer()
-        using var client = GetClient(server.Address);
+        Assert.IsNotNull(server);
+        Assert.IsNotNull(client);
+
+        var accounts = GenerateAccounts();
 
         var okResult = client.CreateAccount(accounts[0]);
         Assert.IsTrue(okResult == CreateAccountResult.Ok);
 
         var lookupAccount = client.LookupAccount(accounts[0].Id);
         Assert.IsNotNull(lookupAccount);
-        AssertAccount(accounts[0], lookupAccount!.Value);
+        AssertAccount(accounts[0], lookupAccount.Value);
 
         var existsResult = client.CreateAccount(accounts[0]);
         Assert.IsTrue(existsResult == CreateAccountResult.Exists);
     }
 
     [TestMethod]
-    [DoNotParallelize]
     public async Task CreateAccountAsync()
     {
-        using var server = new TBServer();
-        using var client = GetClient(server.Address);
+        Assert.IsNotNull(server);
+        Assert.IsNotNull(client);
+
+        var accounts = GenerateAccounts();
 
         var okResult = await client.CreateAccountAsync(accounts[0]);
         Assert.IsTrue(okResult == CreateAccountResult.Ok);
 
         var lookupAccount = await client.LookupAccountAsync(accounts[0].Id);
         Assert.IsNotNull(lookupAccount);
-        AssertAccount(accounts[0], lookupAccount!.Value);
+        AssertAccount(accounts[0], lookupAccount.Value);
 
         var existsResult = await client.CreateAccountAsync(accounts[0]);
         Assert.IsTrue(existsResult == CreateAccountResult.Exists);
     }
 
     [TestMethod]
-    [DoNotParallelize]
     public void CreateAccounts()
     {
-        using var server = new TBServer();
-        using var client = GetClient(server.Address);
+        Assert.IsNotNull(server);
+        Assert.IsNotNull(client);
+
+        var accounts = GenerateAccounts();
 
         var results = client.CreateAccounts(accounts);
         Assert.IsTrue(results.Length == 0);
 
         var lookupAccounts = client.LookupAccounts(new[] { accounts[0].Id, accounts[1].Id });
-        AssertAccounts(lookupAccounts);
+        AssertAccounts(accounts, lookupAccounts);
     }
 
     [TestMethod]
-    [DoNotParallelize]
     public async Task CreateAccountsAsync()
     {
-        using var server = new TBServer();
-        using var client = GetClient(server.Address);
+        Assert.IsNotNull(server);
+        Assert.IsNotNull(client);
+
+        var accounts = GenerateAccounts();
 
         var results = await client.CreateAccountsAsync(accounts);
         Assert.IsTrue(results.Length == 0);
 
         var lookupAccounts = client.LookupAccounts(new[] { accounts[0].Id, accounts[1].Id });
-        AssertAccounts(lookupAccounts);
+        AssertAccounts(accounts, lookupAccounts);
     }
 
     [TestMethod]
-    [DoNotParallelize]
     public void CreateTransfers()
     {
-        using var server = new TBServer();
-        using var client = GetClient(server.Address);
+        Assert.IsNotNull(server);
+        Assert.IsNotNull(client);
```

Comment:
Tbh, I’d drop these extra asserts from tests: this is something forced upon us by the testing framework, but we are using an idiomatic testing framework, and so can accept slight degradation of safety (in tests !!), so as not to swim upstream against C# here.

Reply:
It's due to the compiler null checks. I'd rather use those `Assert.IsNotNull(server)` than the null-forgiving syntax `server!` everywhere.

Reply:
Ahh, that makes sense. Seems like C# has an escape hatch here?

https://stackoverflow.com/a/73167373

Could we declare the relevant fields as non-null and init then with `null!`?